app/src/main/java/io/appium/uiautomator2/server/mjpeg/MjpegScreenshotClient.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.uiautomator2.server.mjpeg;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.util.Locale;
+
+import io.appium.uiautomator2.utils.Logger;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+class MjpegScreenshotClient {
+    private final Socket socket;
+    private boolean closed = false;
+    private boolean initialized = false;
+    private OutputStream out;
+    private BufferedReader in;
+
+    MjpegScreenshotClient(Socket socket) {
+        this.socket = socket;
+    }
+
+    boolean getInitialized() {
+        return initialized;
+    }
+
+    boolean getClosed() {
+        return closed;
+    }
+
+    private String getRemoteAddress() {
+        if (socket == null) {
+            return "";
+        }
+        return socket.getRemoteSocketAddress().toString().replaceAll("^/+", "");
+    }
+
+    void closeSocket() {
+        try {
```

Comment:
we should probably short circuit here if `closed` is already `true`

Reply:
This will be resolved w/ the upcoming update(s)